pkg/ini.go

```diff
@@ -0,0 +1,124 @@
+func (parser *IniParser) LoadFromString(content string) error {
+	var currentSection string
+	lines := strings.Split(content, "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if strings.HasPrefix(line, "[") && strings.HasSuffix(line, "]") {
+			currentSection = line[1 : len(line)-1]
+		} else if strings.Contains(line, "=") && !strings.HasPrefix(line, "#") && !strings.HasPrefix(line, ";") {
+			values := strings.Split(line, "=")
+			err := parser.Set(currentSection, strings.TrimSpace(values[0]), strings.TrimSpace(values[1]))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Load INI content from a file.
+// It could return an error if the file isn't valid or if the file isn't found.
+func (parser *IniParser) LoadFromFile(fileName string) error {
+	dat, err := os.ReadFile(fileName)
+	if err != nil {
+		return fmt.Errorf("file %s not found", fileName)
+	}
+	err = parser.LoadFromString(string(dat))
+	return err
+}
+
+// Retrieve a list of all section names.
+func (parser *IniParser) GetSectionNames() []string {
+	var sections []string
+	for s := range parser.section {
+		sections = append(sections, s)
+	}
+	return sections
+}
+
+// Serialize and convert the INI content into a map with the format {section_name: {key1: val1, key2: val2}, ...}.
+func (parser *IniParser) GetSection() map[string]map[string]string {
+	m2 := make(map[string]map[string]string, len(parser.section))
+	maps.Copy(m2, parser.section)
+	return m2
+}
+
+// Retrieve the value of a key in a specific section.
+// It could return an error if the section/key not found.
+func (parser *IniParser) Get(sectionName, key string) (string, error) {
+	sectionMap, ok := parser.section[sectionName]
+	if !ok {
+		return sectionMap[key], fmt.Errorf("can't get: %s section not found", sectionName)
+	} else if sectionMap[key] == "" {
+		return sectionMap[key], fmt.Errorf("can't get: %s key not found", key)
+	}
+	return sectionMap[key], nil
+}
+
+// Set the value of a key in a specific section.
+// It can also be used to add a new key-value pair in a new section.
+func (parser *IniParser) Set(section, key, value string) error {
+	if section == "" {
+		return errors.New("trying to add key and value for an empty section")
+	} else if key == "" {
+		return fmt.Errorf("can't set/add missing key for section : %s & value : %s", section, value)
+	} else if value == "" {
+		return fmt.Errorf("can't set/add missing value for section : %s & key : %s", section, key)
+	}
+	if parser.section == nil {
+		parser.section = make(map[string]map[string]string)
+	}
+	if parser.section[section] == nil {
+		parser.section[section] = make(map[string]string)
+	}
+	parser.section[section][key] = value
+	return nil
+}
+
+// Convert the INI content back to a string format.
+func (parser *IniParser) String() string {
+	sectionNames := parser.GetSectionNames()
+	var content string
+	for _, section := range sectionNames {
+		content += fmt.Sprintf("\n[%v]\n", section)
```

Comment:
don't add new line before the section, this means an empty line would be added (unnecessarily) to the beginning of the string